pkg/providers/amifamily/resolver.go

```diff
@@ -216,11 +216,17 @@ func (r Resolver) resolveLaunchTemplate(nodeClass *v1beta1.EC2NodeClass, nodeCla
 	if kubeletConfig.MaxPods == nil {
 		kubeletConfig.MaxPods = lo.ToPtr(int32(maxPods))
 	}
+	taints := lo.Flatten([][]core.Taint{
+		nodeClaim.Spec.Taints,
+		nodeClaim.Spec.StartupTaints,
+		{{Key: "karpenter.sh/unregistered", Effect: core.TaintEffectNoExecute}},
```

Comment:
Does this de-dupe?

Reply:
No. Now adding this after in a loop that checks for its existence.